pkg/types/types.go

```diff
@@ -197,29 +197,59 @@ type Metric struct {
 }
 
 // MergeMetrics merges metrics by name.
-func MergeMetrics(metrics [][]Metric) []Metric {
+// It returns merged metrics, number of rendered data points for the returned metrics,
+// and number of inconsistent data points seen (if consistencyCheck is true).
+func MergeMetrics(metrics [][]Metric, consistencyCheck bool) ([]Metric, int, int) {
 	if len(metrics) == 0 {
-		return nil
+		return nil, 0, 0
 	}
 
 	if len(metrics) == 1 {
-		return metrics[0]
+		pointCount := 0
+		ms := metrics[0]
+		for _, m := range ms {
+			pointCount += len(m.Values)
+		}
+		return ms, pointCount, 0
 	}
 
-	names := make(map[string][]Metric)
+	metricByNames := make(map[string][]Metric)
 
 	for _, ms := range metrics {
 		for _, m := range ms {
-			names[m.Name] = append(names[m.Name], m)
+			metricByNames[m.Name] = append(metricByNames[m.Name], m)
 		}
 	}
 
 	merged := make([]Metric, 0)
-	for _, ms := range names {
-		merged = append(merged, mergeMetrics(ms))
+	pointCount := 0
+	inconsistencyCount := 0
```

Comment:
if the rest of the code shortens to `incons` then we could shorten it here as well for consistency

Reply:
please check @timtofan.